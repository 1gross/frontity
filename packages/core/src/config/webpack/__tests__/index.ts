import hash from "hash-it";
import * as path from "path";
import getWebpack from "../";
import { existsSync } from "fs";

jest.mock("path");
const resolve: jest.Mock<typeof path.resolve> = path.resolve as any;
resolve.mockImplementation((_, dir) => dir);

jest.mock("hash-it");
hash.mockImplementation(() => 123);

const babel = {
  development: {
    es5: { presets: ["es5-development"], plugins: [] },
    module: { presets: ["module-development"], plugins: [] },
    node: { presets: ["node-development"], plugins: [] }
  },
  production: {
    es5: { presets: ["es5-production"], plugins: [] },
    module: { presets: ["module-production"], plugins: [] },
    node: { presets: ["node-production"], plugins: [] }
  }
};

test("Babel returns for development", () => {
  expect(
    getWebpack({ mode: "development", babel: babel["development"] })
  ).toMatchSnapshot();
});

test("Babel returns for production", () => {
  expect(
    getWebpack({ mode: "production", babel: babel["production"] })
  ).toMatchSnapshot();
});
