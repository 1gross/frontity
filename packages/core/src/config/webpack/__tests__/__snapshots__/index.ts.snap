// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Babel returns for development 1`] = `
Object {
  "es5": Object {
    "devtool": false,
    "entry": Array [
      "/Users/luisherranz/Code/frontity/packages/core/src/client",
    ],
    "mode": "development",
    "module": Object {
      "rules": Array [
        Object {
          "test": /\\\\\\.\\(j\\|t\\)sx\\?\\$/,
          "use": Object {
            "loader": "babel-loader",
            "options": Object {
              "babelrc": false,
              "plugins": Array [],
              "presets": Array [
                "es5-development",
              ],
            },
          },
        },
      ],
    },
    "name": "es5",
    "output": Object {
      "chunkFilename": "[name].es5.js",
      "filename": "main.es5.js",
      "path": "/Users/luisherranz/Code/frontity/packages/core/build/static",
    },
    "plugins": Array [],
    "resolve": Object {
      "extensions": Array [
        ".js",
        ".jsx",
        ".ts",
        ".tsx",
      ],
    },
    "target": "web",
  },
  "module": Object {
    "devtool": "cheap-module-eval-source-map",
    "entry": Array [
      "webpack-hot-middleware/client",
      "/Users/luisherranz/Code/frontity/packages/core/src/client",
    ],
    "mode": "development",
    "module": Object {
      "rules": Array [
        Object {
          "test": /\\\\\\.\\(j\\|t\\)sx\\?\\$/,
          "use": Object {
            "loader": "babel-loader",
            "options": Object {
              "babelrc": false,
              "plugins": Array [],
              "presets": Array [
                "module-development",
              ],
            },
          },
        },
      ],
    },
    "name": "client",
    "output": Object {
      "chunkFilename": "[name].module.js",
      "filename": "main.module.js",
      "path": "/Users/luisherranz/Code/frontity/packages/core/build/static",
    },
    "plugins": Array [
      LoadablePlugin {
        "compiler": null,
        "handleEmit": [Function],
        "opts": Object {
          "filename": "loadable-stats.json",
          "outputAsset": true,
          "path": undefined,
          "writeToDisk": undefined,
        },
        "writeAssetsFile": [Function],
      },
      HotModuleReplacementPlugin {
        "fullBuildTimeout": 200,
        "multiStep": undefined,
        "options": Object {},
        "requestTimeout": 10000,
      },
    ],
    "resolve": Object {
      "extensions": Array [
        ".js",
        ".jsx",
        ".ts",
        ".tsx",
      ],
    },
    "target": "web",
  },
  "node": Object {
    "devtool": "cheap-module-eval-source-map",
    "entry": Array [
      "/Users/luisherranz/Code/frontity/packages/core/src/server",
    ],
    "mode": "development",
    "module": Object {
      "rules": Array [
        Object {
          "test": /\\\\\\.\\(j\\|t\\)sx\\?\\$/,
          "use": Object {
            "loader": "babel-loader",
            "options": Object {
              "babelrc": false,
              "plugins": Array [],
              "presets": Array [
                "node-development",
              ],
            },
          },
        },
      ],
    },
    "name": "server",
    "output": Object {
      "filename": "main.js",
      "libraryTarget": "commonjs2",
      "path": "/Users/luisherranz/Code/frontity/packages/core/build/dynamic",
    },
    "plugins": Array [
      LoadablePlugin {
        "compiler": null,
        "handleEmit": [Function],
        "opts": Object {
          "filename": "loadable-stats.json",
          "outputAsset": true,
          "path": undefined,
          "writeToDisk": undefined,
        },
        "writeAssetsFile": [Function],
      },
      LimitChunkCountPlugin {
        "options": Object {
          "maxChunks": 1,
        },
      },
    ],
    "resolve": Object {
      "extensions": Array [
        ".js",
        ".jsx",
        ".ts",
        ".tsx",
      ],
    },
    "target": "node",
  },
}
`;

exports[`Babel returns for production 1`] = `
Object {
  "es5": Object {
    "devtool": false,
    "entry": Array [
      "/Users/luisherranz/Code/frontity/packages/core/src/client",
    ],
    "mode": "production",
    "module": Object {
      "rules": Array [
        Object {
          "test": /\\\\\\.\\(j\\|t\\)sx\\?\\$/,
          "use": Object {
            "loader": "babel-loader",
            "options": Object {
              "babelrc": false,
              "plugins": Array [],
              "presets": Array [
                "es5-production",
              ],
            },
          },
        },
      ],
    },
    "name": "es5",
    "output": Object {
      "chunkFilename": "[name].es5.[chunkhash].js",
      "filename": "main.es5.[chunkhash].js",
      "path": "/Users/luisherranz/Code/frontity/packages/core/build/static",
    },
    "plugins": Array [],
    "resolve": Object {
      "extensions": Array [
        ".js",
        ".jsx",
        ".ts",
        ".tsx",
      ],
    },
    "target": "web",
  },
  "module": Object {
    "devtool": false,
    "entry": Array [
      "/Users/luisherranz/Code/frontity/packages/core/src/client",
    ],
    "mode": "production",
    "module": Object {
      "rules": Array [
        Object {
          "test": /\\\\\\.\\(j\\|t\\)sx\\?\\$/,
          "use": Object {
            "loader": "babel-loader",
            "options": Object {
              "babelrc": false,
              "plugins": Array [],
              "presets": Array [
                "module-production",
              ],
            },
          },
        },
      ],
    },
    "name": "client",
    "output": Object {
      "chunkFilename": "[name].module.[chunkhash].js",
      "filename": "main.module.[chunkhash].js",
      "path": "/Users/luisherranz/Code/frontity/packages/core/build/static",
    },
    "plugins": Array [
      LoadablePlugin {
        "compiler": null,
        "handleEmit": [Function],
        "opts": Object {
          "filename": "loadable-stats.json",
          "outputAsset": true,
          "path": undefined,
          "writeToDisk": undefined,
        },
        "writeAssetsFile": [Function],
      },
    ],
    "resolve": Object {
      "extensions": Array [
        ".js",
        ".jsx",
        ".ts",
        ".tsx",
      ],
    },
    "target": "web",
  },
  "node": Object {
    "devtool": false,
    "entry": Array [
      "/Users/luisherranz/Code/frontity/packages/core/src/server",
    ],
    "mode": "production",
    "module": Object {
      "rules": Array [
        Object {
          "test": /\\\\\\.\\(j\\|t\\)sx\\?\\$/,
          "use": Object {
            "loader": "babel-loader",
            "options": Object {
              "babelrc": false,
              "plugins": Array [],
              "presets": Array [
                "node-production",
              ],
            },
          },
        },
      ],
    },
    "name": "server",
    "output": Object {
      "filename": "main.js",
      "libraryTarget": "commonjs2",
      "path": "/Users/luisherranz/Code/frontity/packages/core/build/dynamic",
    },
    "plugins": Array [
      LoadablePlugin {
        "compiler": null,
        "handleEmit": [Function],
        "opts": Object {
          "filename": "loadable-stats.json",
          "outputAsset": true,
          "path": undefined,
          "writeToDisk": undefined,
        },
        "writeAssetsFile": [Function],
      },
      LimitChunkCountPlugin {
        "options": Object {
          "maxChunks": 1,
        },
      },
    ],
    "resolve": Object {
      "extensions": Array [
        ".js",
        ".jsx",
        ".ts",
        ".tsx",
      ],
    },
    "target": "node",
  },
}
`;
